### PR TITLE
Generate changelogs per release

### PR DESCRIPTION
**Pull request by orta**
Add support for re-using the CHANGELOG infra for the releases - https://github.com/microsoft/TypeScript-DOM-lib-generator/releases

### CODE REVIEW

File: src/changelog.ts [hunk 1]
```diff
@@ -169,14 +177,9 @@ export function generate(): string {
   }
 
   const output = outputs.join("\n\n");
-
-  if (!output.length) {
```
Comment (saschanaz) [inline src/changelog.ts:173]: Could you move this logic to the caller?

**Review by saschanaz — APPROVED** [review]
Approving with a nit.

I still think this can be done without unpkg by simply replacing the unpkg part with the `getLatestTag()`, but that's a separate issue.

File: deploy/deployChangedPackages.mjs [hunk 2]
```diff
@@ -93,8 +102,10 @@ Assuming that this means we need to upload this package.`);
 
       uploaded.push(dirName);
     }
-  }
 
+    console.log("\n# Release notes:");
```
Comment (sandersn) [inline deploy/deployChangedPackages.mjs:106]: I don't know the code well enough -- but it looks like this prints the release notes but doesn't upload them. Is that the intent?

Comment (orta) [reply deploy/deployChangedPackages.mjs:106]: It does! I just figured it'd be nice to have that in the CI logs also

**Comment by orta** [comment]
OK, this is updated to use the git tags and all feedback is resolved 👍🏻

File: deploy/createTypesPackages.mjs [hunk 3]
```diff
@@ -131,3 +129,7 @@ function copyREADME(pkg, pkgJSON, writePath) {
 
   fs.writeFileSync(writePath, readme);
 }
+
+if (process.argv[1] === fileURLToPath(import.meta.url)) {
+  go();
```
Comment (saschanaz) [inline deploy/createTypesPackages.mjs:134]:
```suggestion
  await go();
```

Using top level `await` can be helpful as it changes exit code when throws. [It's supported by Node.js 14+.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#browser_compatibility)

File: deploy/deployChangedPackages.mjs [hunk 4]
```diff
@@ -40,28 +44,51 @@ const go = async () => {
...
+      );
+
       const generatedDTSPath = join(generatedDir, dirName, file);
       const generatedDTSContent = fs.readFileSync(generatedDTSPath, "utf8");
-      const unpkgURL = `https://unpkg.com/${pkgJSON.name}/${file}`;
+
+      // This assumes we'll only _ever_ ship patches, which may change in the
+      // future someday.
+      const [maj, min, patch] = pkgJSON.version.split(".");
+      const olderVersion = `${maj}.${min}.${patch - 1}`;
```
Comment (saschanaz) [inline deploy/deployChangedPackages.mjs:72]: (Just to note for future reference: `git describe --tags --match "@types/web@*" --abbrev=0` can do the same thing)

**Review by saschanaz — APPROVED** [review]
Cool!

**Pull request merged**